### PR TITLE
[Cosmos] Migrate get_document to pipelines architecture

### PR DESCRIPTION
**Pull request by eholk**
This migrates `get_document` to the new pipelines architecture and updates all the examples to compile.

I'm sure this can be improved, so I'd definitely appreciate code review feedback. Also, although the examples compile, I didn't run them so I'm not 100% sure this preserves the previous behavior. Is there a standard approach people have to testing these things?

Issue #290

### CODE REVIEW

File: sdk/cosmos/src/operations/get_document.rs [hunk 1]
```diff
@@ -1,38 +1,85 @@
...
+use azure_core::{collect_pinned_stream, Request as HttpRequest, Response as HttpResponse};
 use chrono::{DateTime, Utc};
-use http::response::Response;
-use http::StatusCode;
+use http::{HeaderMap, StatusCode};
 use serde::de::DeserializeOwned;
 
+#[derive(Debug, Clone)]
+pub struct GetDocumentOptions<'a, 'b> {
+    document_client: &'a DocumentClient,
```
Comment (rylev) [inline sdk/cosmos/src/operations/get_document.rs:15]: We don't need the document client or activity id in the options. They can just be removed.

**Review by rylev — APPROVED** [review]
Thanks! Looks great. We don't have a good way of testing right now (that's in the works with #348) so we'll need to test this manually. 

I only had one comment on some dead code we can remove.

Overall, I think it might make sense for us to merge this and then when #348 we can ensure it's in working order.

**Comment by JoshGendein** [comment]
Not sure what others are doing but I just created a cosmos account in a personal subscription and test manually. Also because the e2e tests are behind a feature don't forget to check them with `cargo check --tests --features test_e2e`. I think you can add this as config to rust-analyzer.

File: sdk/cosmos/src/operations/get_document.rs (hunk 1 above)
Comment (eholk) [reply sdk/cosmos/src/operations/get_document.rs]: Ah, thanks. It wasn't clear to me which things were needed and which weren't. I've removed the two now.

**Comment by eholk** [comment]
@JoshGendein - Thanks for the suggestion. I found some more code that needs to be updated by doing that. I'll try doing the e2e tests on a personal subscription too.

**Comment by eholk** [comment]
Okay, I tried running the tests on a personal subscription. I had failures on `create_and_delete_document` as well as `replace_document`, but I get similar failures on the main branch, so it doesn't seem like my change broke it.

**Comment by rylev** [comment]
♥ Thank you!

**Pull request merged**